src/pybind/mgr/dashboard/controllers/nvmeof.py

```diff
@@ -392,23 +395,148 @@ def list(self, nqn: str):
                 NVMeoFClient.pb2.list_connections_req(subsystem=nqn)
             )
 
+    @UIRouter('/nvmeof', Scope.NVME_OF)
+    class NVMeoFTcpUI(BaseController):
+        @Endpoint('GET', '/status')
+        @ReadPermission
+        @EndpointDoc("Display NVMe/TCP service Status",
+                     responses={200: NVME_SCHEMA})
+        def status(self) -> dict:
+            status: Dict[str, Any] = {'available': True, 'message': None}
+            orch_backend = mgr.get_module_option_ex('orchestrator', 'orchestrator')
+            if orch_backend == 'cephadm':
+                orch = OrchClient.instance()
+                orch_status = orch.status()
+                if not orch_status['available']:
+                    return status
+                if not orch.services.list_daemons(daemon_type='nvmeof'):
+                    status["available"] = False
+                    status["message"] = 'Create an NVMe/TCP service to get started.'
+            return status
+
+        @Endpoint('GET', '/subsystems')
+        @ReadPermission
+        def list(self):
+            response = []
+            # Get list of subystems
+            subsystems = NVMeoFClient().stub.list_subsystems(
+                NVMeoFClient.pb2.list_subsystems_req()
+            )
+            if subsystems.status != 0:
+                raise DashboardException(
+                    msg=response.error_message,
+                    code=response.status,
+                    http_status_code=NVMeoFClient().NVMeoFError2HTTP.get(response.status, 400),
+                    component="nvmeof",
+                )
```

Comment:
If you check the `handle_nvmeof_error()` decorator, it's also coping with another error condition, `grpc._channel._InactiveRpcError`. In the future we'll surely face new error conditions, while testing or from user reports. That's why my suggestion is to put error handling under a single code block (and error handler) and not spreading that throughout the codebase.

```suggestion
            # Get list of subystems
            reponse = handle_nvmeof_error(
                NVMeoFClient().stub.list_subsystems(
                    NVMeoFClient.pb2.list_subsystems_req()
                )
            )
```